rising/utils/torchinterp1d.py

```diff
@@ -0,0 +1,204 @@
+                'torchsearchsorted module, which is not available.\n'
+                'You must get it at ',
+                'https://github.com/aliutkus/torchsearchsorted \n')
+
+        # making the vectors at least 2D
+        is_flat = {}
+        require_grad = {}
+        v = {}
+        device = []
+        eps = torch.finfo(y.dtype).eps
+        for name, vec in {'x': x, 'y': y, 'xnew': xnew}.items():
+            assert len(vec.shape) <= 2, 'interp1d: all inputs must be '\
+                                        'at most 2-D.'
+            if len(vec.shape) == 1:
+                v[name] = vec[None, :]
+            else:
+                v[name] = vec
+            is_flat[name] = v[name].shape[0] == 1
+            require_grad[name] = vec.requires_grad
+            device = list(set(device + [str(vec.device)]))
+        assert len(device) == 1, 'All parameters must be on the same device.'
+        device = device[0]
+
+        # Checking for the dimensions
+        assert (v['x'].shape[1] == v['y'].shape[1]
+                and (
+                     v['x'].shape[0] == v['y'].shape[0]
+                     or v['x'].shape[0] == 1
+                     or v['y'].shape[0] == 1
+                    )
+                ), ("x and y must have the same number of columns, and either "
+                    "the same number of row or one of them having only one "
+                    "row.")
+
+        reshaped_xnew = False
+        if ((v['x'].shape[0] == 1) and (v['y'].shape[0] == 1)
+           and (v['xnew'].shape[0] > 1)):
+            # if there is only one row for both x and y, there is no need to
+            # loop over the rows of xnew because they will all have to face the
+            # same interpolation problem. We should just stack them together to
+            # call interp1d and put them back in place afterwards.
+            original_xnew_shape = v['xnew'].shape
+            v['xnew'] = v['xnew'].contiguous().view(1, -1)
+            reshaped_xnew = True
+
+        # identify the dimensions of output and check if the one provided is ok
+        D = max(v['x'].shape[0], v['xnew'].shape[0])
+        shape_ynew = (D, v['xnew'].shape[-1])
+        if out is not None:
+            if out.numel() != shape_ynew[0]*shape_ynew[1]:
+                # The output provided is of incorrect shape.
+                # Going for a new one
+                out = None
+            else:
+                ynew = out.reshape(shape_ynew)
+        if out is None:
+            ynew = torch.zeros(*shape_ynew, device=device)
+
+        # moving everything to the desired device in case it was not there
+        # already (not handling the case things do not fit entirely, user will
+        # do it if required.)
+        for name in v:
+            v[name] = v[name].to(device)
+
+        # calling searchsorted on the x values.
+        ind = ynew.long()
+        searchsorted(v['x'].contiguous(), v['xnew'].contiguous(), ind)
```

Comment:
```suggestion
        torch.searchsorted(v['x'].contiguous(), v['xnew'].contiguous(), ind)
```

Reply:
To be honest I have no clue how torch.searchsorted behaves compared to torchsearchsorted.searchsorted . Please check this yourselves as this is just an "educated" guess.

Reply:
I'll have a look at it...

Reply:
Yep it works, I updated the torchinterp1d file, no need for the old dependency any more thanks to pyTorch 1.6
I updated the other pull request. I guess it would be best if we just have one pull request, instead of two with the same changes?

Reply:
Yes, I agree. 
I think it would be best to have a single PR. I only created a new one, as I could not commit to yours (sorry, I am still a bit confused about github and PRs). I propose that I rebase my additions on your PR again and change the default parameters as follows and according to your proposal https://github.com/PhoenixDL/rising/pull/105#discussion_r510049246: I would like to set all absolute patch sizes to tuples of 0 and only use relative patch_sizes if the absolute patch sizes are tuples of 0 (or each element <= 0). This would result in relative patch sizes being the default and users can overwrite them with absolute patch sizes without having to change relative patch sizes.
I think you need to allow me to commit to your PR though. Let me know what you think :)